### PR TITLE
[PIE-2038] Add new --key-value-storage CLI option.

### PR DESCRIPTION
**Pull request by bgravenorst**
Signed-off-by: Byron Gravenorst <byron.gravenorst@consensys.net>

## PR description
Add new `--key-value-storage` CLI option.

## Fixed Issue(s)
PIE-2038

### CODE REVIEW

File: docs/Reference/CLI/CLI-Syntax.md [hunk 1]
```diff
@@ -275,6 +275,26 @@ By default, access from `localhost` and `127.0.0.1` is accepted.
 !!!tip
     To allow all hostnames, use `"*"`. We don't recommend allowing all hostnames for production code.
 
+### key-value-storage
+
+```bash tab="Syntax"
+--key-value-storage=<keyValueStorageName>
```
Comment (pinges) [inline docs/Reference/CLI/CLI-Syntax.md]: default is "rocksdb"

File: docs/Reference/CLI/CLI-Syntax.md [hunk 2]
```diff
@@ -275,6 +275,26 @@ By default, access from `localhost` and `127.0.0.1` is accepted.
...
+
+```bash tab="Environment Variable"
+BESU_KEY_VALUE_STORAGE=rocksdb
+```
+
+```bash tab="Configuration File"
+key-value-storage="rocksdb"
+```
+
+Key-value storage to be used. Use this option only if using a storage system provided with a plugin. Defaults to `rocksdb`.
```
Comment (MadelineMurray) [inline docs/Reference/CLI/CLI-Syntax.md]: Defaults to -> Default is 

Suggested change to be consistent with other CLI options

Comment (bgravenorst) [reply docs/Reference/CLI/CLI-Syntax.md]: Done.

File: docs/Reference/CLI/CLI-Syntax.md (hunk 1 above)
Comment (bgravenorst) [reply docs/Reference/CLI/CLI-Syntax.md]: Done

**Pull request merged**